test/modules/packages/distributedIters/DistributedIters.chpl

```diff
@@ -0,0 +1,398 @@
+      var newRange = t(1);
+
+      type cType = c.type;
+      // Does the same thing as densify, but densify makes a stridable domain,
+      // which mismatches here if c (and thus cType) is non-stridable.
+      var tempDom : cType = computeZeroBasedDomain(c);
+
+      // Rank-change slice the domain along parDim
+      var tempTup = tempDom.dims();
+      // Change the value of the parDim elem of the tuple to the new range
+      tempTup(parDim) = newRange;
+
+      yield tempTup;
+    }
+  }
+  else // c is a range.
+  {
+    const iterCount = c.length;
+
+    if iterCount == 0 then halt("DistributedIters: Guided iterator (leader): ",
+                                "the range is empty");
+
+    type cType = c.type;
+    const denseRange:cType = densify(c,c);
+
+    if iterCount == 1
+       || numTasks == 1 && numLocales == 1
+    then
+    {
+      if debugDistributedIters
+      then writeln("DistributedIters: Guided iterator (leader): serial ",
+                   "execution due to insufficient work or compute resources");
+      yield (denseRange,);
+    }
+    else
+    {
+      const numWorkerLocales = workerLocales.size;
+      const denseRangeHigh:int = denseRange.high;
+      const masterLocale = here.locale;
+
+      const potentialWorkerLocales =
+        [L in workerLocales] if numLocales == 1
+                                || !coordinated
+                                || L != masterLocale
+                             then L;
+      /*
+        It's not sensible to use a single locale besides masterLocale, so use
+        potentialWorkerLocales only if it's larger than one locale.
+      */
+      const actualWorkerLocales = if potentialWorkerLocales.size > 1
+                                  then potentialWorkerLocales
+                                  else [masterLocale];
+      const numActualWorkerLocales = actualWorkerLocales.size;
+
+      // The guided iterator stage (determines next subrange index and size).
+      var meitneriumIndex:atomic int;
+
+      if infoDistributedIters then
+      {
+        const actualWorkerLocaleIds = [L in actualWorkerLocales] L.id:string;
+        const actualWorkerLocaleIdsSorted = actualWorkerLocaleIds.sorted();
+        writeln("DistributedIters: guidedDistributed:");
+        writeln("  coordinated = ", coordinated);
+        writeln("  numLocales = ", numLocales);
+        writeln("  numWorkerLocales = ", numWorkerLocales);
+        writeln("  numActualWorkerLocales = ", numActualWorkerLocales);
+        writeln("  masterLocale.id = ", masterLocale.id);
+        writeln("  actualWorkerLocaleIds = [ ",
+                ", ".join(actualWorkerLocaleIdsSorted),
+                " ]");
+      }
+
+      var localeTimes:[0..#numLocales]real;
+      var totalTime:Timer;
+      if timeDistributedIters then totalTime.start();
+
+      coforall L in actualWorkerLocales
+      with (ref meitneriumIndex, ref localeTimes) do
+      on L do
+      {
```

Comment:
There are a few places in this file that have a ``do`` followed by a ``{``, which is unnecessary.